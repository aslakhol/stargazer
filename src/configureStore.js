import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers/rootReducer';

/* eslint-disable no-underscore-dangle */
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const enhancer = composeEnhancers(
  applyMiddleware(thunkMiddleware),
);
/* eslint-enable */

export default function configureStore(preloadedState) {
  return createStore(
    rootReducer,
    preloadedState,
    enhancer,
  );
}
