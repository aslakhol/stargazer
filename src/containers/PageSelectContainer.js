import { connect } from 'react-redux';

import PageSelect from '../components/PageSelect';

const mapStateToProps = state => ({
});

const mapDispatchToProps = dispatch => ({

});

const PageSelectContainer = connect(
  mapStateToProps, mapDispatchToProps,
)(PageSelect);

export default PageSelectContainer;
