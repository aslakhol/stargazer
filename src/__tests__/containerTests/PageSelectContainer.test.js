import React from 'react';
import { PaginationItem, PaginationLink } from 'reactstrap';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import PageSelectContainer from '../../containers/PageSelectContainer';

const middleWares = [thunk];
const mockStore = configureStore(middleWares);
const initialState = {
  request: {
    count: 10,
    currentPage: 1,
  },
};
const store = mockStore(initialState);

describe('<PageSelectContainer />', () => {
  it('should render successfully', () => {
    const component = shallow(<PageSelectContainer store={store} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Clicking the buttons', () => {
    describe('first button', () => {
      it('is disabled', () => {
        const wrapper = shallow(<PageSelectContainer store={store} request={[]} />);
        const component = wrapper.dive();

        const paginationItem = component.find(PaginationItem).first();

        expect(toJson(paginationItem).props.disabled).toBe(false);
      });
    });

    describe('not first nor last button', () => {
      it('dispatches go to page', () => {
        const wrapper = shallow(<PageSelectContainer store={store} request={[]} />);
        const component = wrapper.dive();

        const paginationItem = component.find(PaginationItem).at(1);

        paginationItem.props().onClick();
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    describe('last button', () => {
      it('dispatches go to next', () => {
        const wrapper = shallow(<PageSelectContainer store={store} request={[]} />);
        const component = wrapper.dive();

        const paginationItem = component.find(PaginationLink).last();

        paginationItem.props().onClick();
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });
});
