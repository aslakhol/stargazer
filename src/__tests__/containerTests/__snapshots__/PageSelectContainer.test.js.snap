// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageSelectContainer /> Clicking the buttons last button dispatches go to next 1`] = `
Array [
  Object {
    "pageNum": 0,
    "type": "SET_PAGE",
  },
  Object {
    "timeout": 6,
    "type": "NEW_TIMEOUT",
  },
  Object {
    "type": "NEXT_PAGE",
  },
  Object {
    "timeout": 9,
    "type": "NEW_TIMEOUT",
  },
]
`;

exports[`<PageSelectContainer /> Clicking the buttons not first nor last button dispatches go to page 1`] = `
Array [
  Object {
    "pageNum": 0,
    "type": "SET_PAGE",
  },
  Object {
    "timeout": 6,
    "type": "NEW_TIMEOUT",
  },
]
`;

exports[`<PageSelectContainer /> should render successfully 1`] = `
<PageSelect
  count={10}
  goToPage={[Function]}
  nextPage={[Function]}
  pageCount={3}
  prevPage={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
